lib/ansible/utils/encrypt.py

```diff
@@ -72,6 +72,14 @@ class BaseHash(object):
     def __init__(self, algorithm):
         self.algorithm = algorithm
 
+    def _clean_salt(self, salt, has_raw_salt=False):
+        if not salt:
+            return None
+        elif has_raw_salt:
+            return to_bytes(salt, encoding='ascii', errors='strict')
+        else:
+            return to_native(to_text(salt, encoding='ascii', errors='strict'))
```

Comment:
```suggestion
            return to_native(salt, encoding='ascii', errors='strict')
```

Reply:
Could the salt consist of non-ASCII characters? Seems like this could break things pretty easily if the user provides the salt.

Reply:
Some algorithms in passlib may accept non-ascii chars. But right now ansible doesn't accept non-ascii salt for passlib and @mfuchs left the comment in the component - "passlib salt values must be pure ascii ... ". Not sure what was the reason (probably its about how ansible handle non-ascii chars provided by user in general), so i decided to not break it for now. All algorithms in crypt module don't accept non-ascii salt (https://docs.python.org/2.7/library/crypt.html#crypt.crypt), set of chars actually is much smaller than ascii, thus another problem is that we dont check if user-provided salt is acceptable by particular algorithm, i'll create another PR to fix that (and someone should definitely revisit this component for massive refactoring)